test-jest/cartesian/Scatter.spec.tsx

```diff
@@ -0,0 +1,108 @@
+      <Surface width={500} height={500}>
+        <Scatter points={[] as any} />
+      </Surface>,
+    );
+
+    expect(container.querySelectorAll('.recharts-scatter-symbol')).toHaveLength(0);
+  });
+
+  test('Render customized symbols when shape is set to be a ReactElement', () => {
+    const CustomizedShape = ({ cx, cy }: { cx: number; cy: number }) => (
+      <circle cx={cx} cy={cy} r={5} className="customized-shape" />
+    );
+    const { container } = render(
+      <Surface width={500} height={500}>
+        <Scatter isAnimationActive={false} shape={<CustomizedShape cx={0} cy={0} />} points={data as any} />
+      </Surface>,
+    );
+
+    expect(container.querySelectorAll('.customized-shape')).toHaveLength(data.length);
+  });
+
+  test('Render customized symbols when shape is set to be a function', () => {
+    const renderCustomizedShape = ({ cx, cy }: { cx: number; cy: number }) => (
+      <circle cx={cx} cy={cy} r={5} className="customized-shape" />
+    );
+    const { container } = render(
+      <Surface width={500} height={500}>
+        <Scatter isAnimationActive={false} shape={renderCustomizedShape} points={data as any} />
+      </Surface>,
+    );
+
+    expect(container.querySelectorAll('.customized-shape')).toHaveLength(data.length);
+  });
+
+  test('Render customized line when line is set to be a ReactElement', () => {
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars, react/prop-types
+    const CustomizedLine = ({ points }: { points: any }) => <path d="M0,0L200,200" className="customized-line" />;
+    const { container } = render(
+      <Surface width={500} height={500}>
+        <Scatter isAnimationActive={false} line={<CustomizedLine points={data as any} />} points={data as any} />
+      </Surface>,
+    );
+
+    expect(container.querySelectorAll('.customized-line')).toHaveLength(1);
+  });
+
+  test('Render customized line when line is set to be a function', () => {
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const renderCustomizedLine = ({ points }: { points: any }) => <path d="M0,0L200,200" className="customized-line" />;
+    const { container } = render(
+      <Surface width={500} height={500}>
+        <Scatter isAnimationActive={false} line={renderCustomizedLine} points={data as any} />
+      </Surface>,
+    );
+
+    expect(container.querySelectorAll('.customized-line')).toHaveLength(1);
+  });
+
+  test('mouse enter or mouse leave a symbol', () => {
+    const onClick = jest.fn();
+    const onMouseEnter = jest.fn();
+    const onMouseLeave = jest.fn();
+
+    const { container } = render(
+      <Surface width={500} height={500}>
+        <Scatter
+          isAnimationActive={false}
+          points={data as any}
+          onClick={onClick}
+          onMouseEnter={onMouseEnter}
+          onMouseLeave={onMouseLeave}
+        />
+      </Surface>,
+    );
+    const syb = container.querySelectorAll('.recharts-symbols')[4];
```

Comment:
Please do not use abbreviations.
```suggestion
    const symbols = container.querySelectorAll('.recharts-symbols')[4];
```